src/MediatR/Internal/RequestHandlerWrapper.cs

```diff
@@ -64,7 +66,11 @@ internal class RequestHandlerWrapperImpl<TRequest, TResponse> : RequestHandlerWr
             return serviceFactory
                 .GetInstances<IPipelineBehavior<TRequest, TResponse>>()
                 .Reverse()
-                .Aggregate((RequestHandlerDelegate<TResponse>) Handler, (next, pipeline) => () => pipeline.Handle((TRequest)request, cancellationToken, next))();
+                .OrderBy(i => i.GetType()
+                                  .GetCustomAttributes(typeof(PipelinePriorityAttribute), true)
```

Comment:
So are you suggesting the only way to get priority is through an Attribute? not adding it to an interface?

Reply:
> So are you suggesting the only way to get priority is through an Attribute? got any thoughts on adding it to an interface?

I thought about doing it via an interface but then there is no way to override. (i.e. In an inheritance scenario if the child class sets a different priority to the parent. With an attribute the parent's one get overridden.)

The priority attribute is optional so it doesn't break any existing code and honors the order of registration in the DI container.

> I wouldn't mind seeing some debug function we could call to print out the priority along with type name and full name, like you could do with structuremaps WhatDoIHave, WhatDidIScan - this just may help with debugging stuff...

I can certainly do that but we would need to do that per `request type` because that's what's how MediatR resolves `IRequestHandler<TRequest, TResult>` and middleware.